app.js

```diff
@@ -48,10 +37,38 @@ angular.module('PathOfDamage', [])
         }
       }
     }
-    $scope.updateHits(true);
-    if (!skipStringify) {
-      $scope.stringifyUrlData();
+    if (!skipUpdates) {
+      $scope.updateHits();
+    }
+  };
+
+  $scope.updateHits = function (skipSerialization) {
+    for (var i = 0; i < $scope.hits.length - 1; i++) {
+      $scope.hits[i] = calcDamage($scope.hits[i].hit);
+    }
+    if (!skipSerialization) {
+      serializeData();
+    }
+  };
+
+  $scope.updateHitTable = function (index) {
+    if ($scope.hits[index].hit === null) {
+      $scope.hits.splice(index, 1);
+      return;
+    }
+    if ($scope.hits[$scope.hits.length - 1].hit !== null) {
+      $scope.hits.push({hit: null});
+    }
+    $scope.hits[index] = calcDamage($scope.hits[index].hit);
+    serializeData();
+  };
+
+  $scope.getHitWidth = function (damage) {
+    var percent = (damage / $scope.healthPool) * 100;
+    if (percent > 100) {
+      percent = 100;
     }
+    return {width: (percent || 0) + "%"};
   };
 
   function calcDamage(hit) {
```

Comment:
I think we want to only check if the hit is null, if it's 0 we want to go ahead and do the calculations right?